source/extensions/common/wasm/context.cc

```diff
@@ -1233,34 +1233,32 @@ WasmResult Context::defineMetric(uint32_t metric_type, std::string_view name,
   if (metric_type > static_cast<uint32_t>(MetricType::Max)) {
     return WasmResult::BadArgument;
   }
-  auto type = static_cast<MetricType>(metric_type);
-  // TODO: Consider rethinking the scoping policy as it does not help in this case.
-  Stats::StatNameManagedStorage storage(std::string(name), wasm()->scope_->symbolTable());
-  Stats::StatName stat_name = storage.statName();
+  const auto type = static_cast<MetricType>(metric_type);
+  const Stats::DynamicName dynamic_name = Stats::DynamicName(toAbslStringView(name));
```

Comment:
drop this line.

Going in this direction is can expand memory as they'll be no sharing of symbols with the same stat name across different scopes.

This is an easy way out of contention issues by expanding memory use. Let's not have this be the only way to create stats in wasm.